front-end/studio/src/app/pages/apis/{apiId}/api-detail.page.html

```diff
@@ -69,25 +78,31 @@ <h1 [title]="apiIconTitle()" class="card-pf-title">
                                                         <span>Publish...</span>
                                                     </a>
                                                 </li>
-                                                <li *ngIf="config.isMicrocksEnabled() && isOpenApi30()">
+                                                <li *ngIf="actionEnabled('mock')">
                                                     <a [routerLink]="[ 'mock' ]">
                                                         <span class="fa fa-fw fa-cloud-upload"></span>
                                                         <span>Mock with Microcks...</span>
                                                     </a>
                                                 </li>
                                                 <li role="separator" class="divider"></li>
-                                                <li *ngIf="actionEnabled('download')">
+                                                <li *ngIf="actionEnabled('download-yaml')">
                                                     <a href="{{ uiUrl }}download?type=api&format=yaml&id={{ api.id }}" download="{{ api.name }}.yaml">
                                                         <span class="fa fa-fw fa-download"></span>
                                                         <span>Download (YAML)</span>
                                                     </a>
                                                 </li>
-                                                <li *ngIf="actionEnabled('download')">
+                                                <li *ngIf="actionEnabled('download-json')">
                                                     <a href="{{ uiUrl }}download?type=api&format=json&id={{ api.id }}" download="{{ api.name }}.json">
                                                         <span class="fa fa-fw fa-download"></span>
                                                         <span>Download (JSON)</span>
                                                     </a>
                                                 </li>
+                                                <li *ngIf="actionEnabled('download-sdl')">
+                                                    <a href="{{ uiUrl }}download?type=api&format=sdl&id={{ api.id }}" download="{{ api.name }}.graphql">
+                                                        <span class="fa fa-fw fa-download"></span>
+                                                        <span>Download (SDL)</span>
```

Comment:
Side note - GraphQL will have 2 forms:

- Human readable SDL language
- Introspection.json file 

SQL is used in server-side
Introspection is used in client-side generators.
While SDL is perfect here there can be separate issue to download introspection that community can do.

Reply:
Yeah I have a TODO in Apicurio to support the JSON format, but I didn't immediately see how to get a JSON formatted document out of the `graphql-java` library I'm using.